src/lightning_utilities/testing/runif.py

```diff
@@ -0,0 +1,119 @@
+# Copyright The PyTorch Lightning team.
+# Licensed under the Apache License, Version 2.0 (the "License");
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+import operator
+import os
+import platform
+import sys
+from typing import Any, Optional
+
+from packaging.version import Version
+
+from lightning_utilities.core.imports import compare_version, module_available
+
+if module_available("_pytest"):
+    from _pytest.mark import MarkDecorator
+else:
+    MarkDecorator = None
+
+
+class RunIf:
+    """RunIf wrapper for simple marking specific cases, fully compatible with `pytest.mark`.
+
+    Example:
+        @RunIf(min_torch="0.0")
+        @pytest.mark.parametrize("arg1", [1, 2.0])
+        def test_wrapper(arg1):
+            assert arg1 > 0.0
+    """
+
+    def __new__(  # type: ignore[misc]
+        cls,
+        *args: Any,
+        min_torch: Optional[str] = None,
+        max_torch: Optional[str] = None,
+        min_python: Optional[str] = None,
+        skip_windows: bool = False,
+        standalone: bool = False,
+        min_cuda_gpus: int = 0,
+        is_mps_gpu: Optional[bool] = None,
+        **kwargs: Any,
```

Comment:
We've discussed this several times already :cry:
RunIf cannot be added as it is package-dependent. Only min/max_python and skip_windows could be

Reply:
Do you mean in lightning?

Reply:
yes

Reply:
yes

Reply:
I would suggest `utilities/testing.py`, but yes. That is a better way

Reply:
well, it is already in utilities/testing/runif with optional import for all so zero base dependencies...

Reply:
It's not just about whether the dependency is installed or not. This torch-specific logic does not belong here. It's like adding a Trainer to torchmetrics

Reply:
so then I'll need to create a new package `utilities-useful and use it in almost all projects

Reply:
I'm sorry but I don't understand. I said that `lightning/src/pytorch/utilities/testing.py` was fine. What is the issue with that?